server/src/data/listings.ts

```diff
@@ -12,6 +14,46 @@ export const getListing = async (listingId: string) => {
   return listing.data();
 };
 
+export const getAllListings = async (pageNum: number) => {
+  let listingLimit: number = pageNum * itemsPerPage;
+  const first = query(
+    collection,
+    orderBy("averageRating"),
+    limit(listingLimit)
+  );
+  const snapshot = await firestore.getCountFromServer(collection);
+  let totalListingsCount: number = snapshot.data().count;
+  const documentSnapshots = await getDocs(first);
+  let docsReturnedCount: number = documentSnapshots.size;
+  if (listingLimit - 10 > docsReturnedCount) {
+    let returnObj = {
+      next: null,
+      data: [],
+    };
+    return returnObj;
+  }
+  let listingsDoc;
+  if (listingLimit > docsReturnedCount) {
+    let itemsToReturn = itemsPerPage - (listingLimit - docsReturnedCount);
+    listingsDoc = documentSnapshots.docs.slice(-itemsToReturn);
+  } else {
+    listingsDoc = documentSnapshots.docs.slice(-itemsPerPage);
+  }
+  let listings = [];
+  listingsDoc.forEach((doc) => {
+    listings.push(doc.data());
+  });
+  let nextURL: string =
+    totalListingsCount === docsReturnedCount
+      ? null
+      : `/listings/page/${pageNum + 1}`;
+  let returnObj = {
+    next: nextURL,
```

Comment:
Nothing wrong. getAllListing conflicts with mine which is okay because i can do pagination like this as well. I had other ideas for pagination but to keep its consistent we can go with this. One thing if you can change the returnObj next field to just be ${pageNum + 1} instead of `/listings/page/${pageNum + 1}` so it can be reused